lib/src/HttpUtils.cc

```diff
@@ -684,4 +553,180 @@ FileType parseFileType(const string_view &fileExtension)
+            return sv;
+        }
+        case CT_IMAGE_SVG_XML:
+        {
+            static string_view sv = "image/svg+xml";
+            return sv;
+        }
+        case CT_APPLICATION_X_FONT_TRUETYPE:
+        {
+            static string_view sv = "application/x-font-truetype";
+            return sv;
+        }
+        case CT_APPLICATION_X_FONT_OPENTYPE:
+        {
+            static string_view sv = "application/x-font-opentype";
+            return sv;
+        }
+        case CT_APPLICATION_FONT_WOFF:
+        {
+            static string_view sv = "application/font-woff";
+            return sv;
+        }
+        case CT_APPLICATION_FONT_WOFF2:
+        {
+            static string_view sv = "application/font-woff2";
+            return sv;
+        }
+        case CT_APPLICATION_VND_MS_FONTOBJ:
+        {
+            static string_view sv = "application/vnd.ms-fontobject";
+            return sv;
+        }
+        case CT_APPLICATION_PDF:
+        {
+            static string_view sv = "application/pdf";
+            return sv;
+        }
+        case CT_IMAGE_PNG:
+        {
+            static string_view sv = "image/png";
+            return sv;
+        }
+        case CT_IMAGE_AVIF:
+        {
+            static string_view sv = "image/avif";
+            return sv;
+        }
+        case CT_IMAGE_WEBP:
+        {
+            static string_view sv = "image/webp";
+            return sv;
+        }
+        case CT_IMAGE_JPG:
+        {
+            static string_view sv = "image/jpeg";
+            return sv;
+        }
+        case CT_IMAGE_GIF:
+        {
+            static string_view sv = "image/gif";
+            return sv;
+        }
+        case CT_IMAGE_XICON:
+        {
+            static string_view sv = "image/x-icon";
+            return sv;
+        }
+        case CT_IMAGE_BMP:
+        {
+            static string_view sv = "image/bmp";
+            return sv;
+        }
+        case CT_IMAGE_ICNS:
+        {
+            static string_view sv = "image/icns";
+            return sv;
+        }
+        case CT_APPLICATION_WASM:
+        {
+            static string_view sv = "application/wasm";
+            return sv;
+        }
+        case CT_NONE:
+        {
+            static string_view sv = "";
+            return sv;
+        }
+        default:
+        case CT_TEXT_PLAIN:
+        {
+            static string_view sv = "text/plain; charset=utf-8";
+            return sv;
+        }
+    }
+}
+
+void registerCustomExtensionMime(const std::string &ext,
+                                 const std::string &mime)
+{
+    if (ext.empty())
+        return;
+    auto &mimeStr = customMime[ext];
+    if (!mimeStr.empty())
+    {
+        LOG_WARN << ext << " has already been registered as type " << mime
+                 << ". Overwriting.";
+    }
+    mimeStr = mime;
+}
+
+std::string fileNameToMime(const std::string &fileName)
```

Comment:
it should be better to return a string_view here to avoid constructing a string object.